tests/unit/test-firefox/test.firefox.js

```diff
@@ -545,6 +646,136 @@ describe('firefox', () => {
+        sinon.match(userDirectoryPath),
+      );
+    });
+
+    it('returns a finder that looks for a default profile', async () => {
+      const FxProfile = {
+        Finder: sinon.spy(() => () => Promise.resolve({})),
+      };
+      const fakeReadProfiles = sinon.spy(() => Promise.resolve());
+      const fakeGetPath = sinon.spy(() => Promise.resolve());
+      FxProfile.Finder.prototype.readProfiles = fakeReadProfiles;
+      FxProfile.Finder.prototype.getPath = fakeGetPath;
+      FxProfile.Finder.prototype.profiles = [{
+        Name: 'someName',
+      }];
+
+      const userDirectoryPath = '/non/existent/path';
+
+      const getter = firefox.defaultCreateProfileFinder({
+        userDirectoryPath,
+        FxProfile,
+      });
+
+      await getter('someName');
+
+      sinon.assert.called(fakeReadProfiles);
+      sinon.assert.called(fakeGetPath);
+      sinon.assert.calledWith(
+        fakeGetPath,
+        sinon.match('someName'),
+      );
+    });
+
+    it('returns a finder that resolves undefined for no profiles.ini',
+       async () => {
+         const FxProfile = {
+           Finder: sinon.spy(() => () => Promise.resolve({})),
+         };
+
+         const fakeReadProfiles = sinon.spy(() => {
+           return Promise.reject(
+             new ErrorWithCode('ENOENT', 'fake ENOENT error'));
+         });
+         const fakeGetPath = sinon.spy(() => Promise.resolve());
+         FxProfile.Finder.prototype.readProfiles = fakeReadProfiles;
+         FxProfile.Finder.prototype.getPath = fakeGetPath;
+         FxProfile.Finder.prototype.profiles = [{
+           Name: 'someName',
+         }];
+
+         const userDirectoryPath = '/non/existent/path';
+
+         const getter = firefox.defaultCreateProfileFinder({
+           userDirectoryPath,
+           FxProfile,
+         });
+
+         const res = await getter('someName');
+         assert.equal(
+           res,
+           undefined,
+           'Got an undefined result when the profiles.ini file does not exist');
+
+         sinon.assert.called(fakeReadProfiles);
+         sinon.assert.notCalled(fakeGetPath);
+       });
+
+    it('returns a finder that throws other errors',
+       async () => {
+         const FxProfile = {
+           Finder: sinon.spy(() => () => Promise.resolve({})),
+         };
+
+         const fakeReadProfiles = sinon.spy(() => {
+           return Promise.reject(
+             new Error('unspecified error'));
+         });
+         const fakeGetPath = sinon.spy(() => Promise.resolve());
+         FxProfile.Finder.prototype.readProfiles = fakeReadProfiles;
+         FxProfile.Finder.prototype.getPath = fakeGetPath;
+         FxProfile.Finder.prototype.profiles = [{
+           Name: 'someName',
+         }];
+
+         const userDirectoryPath = '/non/existent/path';
+
+         const getter = firefox.defaultCreateProfileFinder({
+           userDirectoryPath,
+           FxProfile,
+         });
+
+         try {
```

Comment:
Let's rewrite this using `assert.isRejected` and the approach described in one of the comments above.

(as a side note, when we use the `try...catch` strategy to assert a raised exception, we always have to be sure that the test fails if the expected exception has not be raised, e.g. by just get a reference to the actual error from the catch block and then call the assertion after the `try...catch`).